### PR TITLE
Fix broken links in docs

### PR DESCRIPTION
**Pull request by rogr**
Fix all broken links on docs.

### CODE REVIEW

File: docs/testing/component-testing.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ function Button(props) {
 export default Button;
 ```
 
-> Note: This is a [stateless ("dumb") component](../js/react/README.md#components-and-containers)!
+> Note: This is a [stateless ("dumb") component](../js/README.md#architecture-components-and-containers)!
```
Comment (rogr) [inline docs/testing/component-testing.md]: I'm not sure if this the correct link.

File: docs/testing/component-testing.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ class HomePage extends React.Component {
 }
 ```
 
-> Note: This is a [stateful ("smart") component](../js/react/README.md#components-and-containers)!
+> Note: This is a [stateful ("smart") component](../js/README.md#architecture-components-and-containers)!
```
Comment (rogr) [inline docs/testing/component-testing.md]: I'm not sure if this the correct link.

**Comment by mxstbr** [comment]
Awesome, thanks so much!!

File: docs/testing/component-testing.md (hunk 2 above)
Comment (mxstbr) [inline docs/testing/component-testing.md]: Yeah it is! :100:

**Pull request merged**